dash-pipeline/bmv2/dash_outbound.p4

```diff
@@ -47,14 +32,14 @@ control outbound(inout headers_t hdr,
 
     action set_tunnel_mapping(IPv4Address underlay_dip,
                               EthernetAddress overlay_dmac,
-                              bit<1> use_dst_vni) {
+                              bit<1> use_dst_vnet) {
         /*
-           if (use_dst_vni)
+           if (use_dst_vnet)
                vni = meta.encap_data.vni;
           else
               vni = meta.encap_data.dest_vnet_vni;
         */
-        meta.encap_data.vni = meta.encap_data.vni * (bit<24>)(~use_dst_vni) + meta.encap_data.dest_vnet_vni * (bit<24>)use_dst_vni;
+        meta.vnet_id = meta.vnet_id * (bit<16>)(~use_dst_vnet) + meta.dst_vnet_id * (bit<16>)use_dst_vnet;
```

Comment:
did you try the simple way - did this not compile ?
```
if (use_dst_vnet) {
    meta.vnet_id = meta.dst_vnet_id;
}

```

Reply:
didnt try, retain the old logic ?

Reply:
yeah we can keep that if it compiles .. easier to understand